pkg/resourceprovider/provider.go

```diff
@@ -120,6 +125,45 @@ type nsQueryResults struct {
 	err       error
 }
 
+var (
+	queryFailureCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "namespace_query_failure_total",
+			Help: "Total number of failed namespace query attempts in GetPodMetrics, labeled by namespace and error message",
+		},
+		[]string{"namespace", "error"},
+	)
+)
+
+func init() {
+	mux := http.NewServeMux()
+
+	mux.Handle("/metrics", promhttp.InstrumentMetricHandler(
+		prometheus.DefaultRegisterer,
+		promhttp.HandlerFor(prometheus.DefaultGatherer, promhttp.HandlerOpts{
+			ErrorHandling: promhttp.PanicOnError,
+		}),
+	))
+
+	// Register the metric with Prometheus.
+	prometheus.MustRegister(queryFailureCounter)
+
+	// Create the listener manually
+	listener, err := net.Listen("tcp", ":8080")
+	if err != nil {
+		klog.Fatalf("[http] Failed to create listener: %+v", err)
+	}
+	klog.Infof("[http] listening on %s", listener.Addr())
+
+	// Start serving using the listener
+	go func() {
+		err := http.Serve(listener, mux)
+		if err != nil {
+			klog.Warningf("[http] error serving http: %+v", err)
+		}
```

Comment:
Needs more descriptive error messages that these are to handle prom metrics scrapes